craft_parts/executor/environment.py

```diff
@@ -140,6 +140,35 @@ def _basic_environment_for_part(part: Part, *, step_info: StepInfo) -> Dict[str,
     return part_environment
 
 
+def _get_step_environment(step_info: StepInfo) -> Dict[str, str]:
+    """Set environment variables from step information.
+
+    Variable names are prefixed by the application name and follows the
+    list of environment variables set by Snapcraft.
+
+    :param step_info: Information about the current step.
+
+    :return: A dictionary containing environment variables and values.
+    """
+    # The application name contains only alphanumeric characters or underscore.
```

Comment:
is this being checked for?

Reply:
it is, further bellow, maybe this requires a new check as the attribute is mutable?